src/otx/core/types/label.py

```diff
@@ -279,8 +317,60 @@ def convert_labels_if_needed(
             label_to_idx=label_to_idx,
             label_tree_edges=get_label_tree_edges(dm_label_categories.items),
             empty_multiclass_head_indices=[],  # consider the label removing case
+            label_ids=[str(i) for i in range(len(label_names))],
         )
 
+    @classmethod
+    def from_dm_label_groups_arrow(cls, dm_label_categories: LabelCategories) -> HLabelInfo:
+        """Generate HLabelData from the Datumaro LabelCategories. Arrow-specific implementation.
+
+        Args:
+            dm_label_categories (LabelCategories): the label categories of datumaro.
+        """
+        dm_label_categories = copy.deepcopy(dm_label_categories)
+
+        empty_label_name = None
```

Comment:
I think we need to remove empty_label for other tasks also. Can you add similar approach to the base LabelInfo?

Reply:
It's too dangerous for 2.6, I'm even going to reduce the set of changes in this PR. The idea of the PR is to delegate further filtering to task-specific datasets instead of trying to filter in one place. However, now I'm not sure if this approach is the best option. H-cls can not be filtered by datumaro's standard methods, so the related changes for h-labels would stay.